coalib/output/ConsoleInteraction.py

```diff
@@ -490,27 +490,26 @@ def choose_action(console_printer, actions):
     :param actions:         Actions available to the user.
     :return:                Return choice of action of user.
     """
+    print_colors = ["red", "yellow", "green", "cyan"]
     console_printer.print(format_lines(
         "The following actions are applicable to this result:"))
 
     while True:
-        console_printer.print(format_lines(" 0: " +
-                                           "Apply no further actions."))
+        console_printer.print(format_lines("(e)xit."), color=print_colors[0])
         for i, action in enumerate(actions, 1):
-            console_printer.print(format_lines("{:>2}: {}".format(
-                i,
-                action.desc)))
+            console_printer.print(format_lines(
+                action.desc), color=print_colors[i])
 
         try:
-            line = format_lines("Please enter the number of the action "
-                                "you want to execute. ")
-            choice = int(input(line))
-            if 0 <= choice <= len(actions):
+            line = format_lines("Please enter the alphabet for the action "
+                                "you want to execute: ")
+            choice = input(line)
+            if choice in {'e', 'o', 'a', 's'}:
```

Comment:
yeah something like that, please discuss the design on the issue, this is not solved with something as trivial as this